src/main.rs

```diff
@@ -7,9 +7,19 @@ use anyhow::Result;
 
 fn main() -> Result<()> {
     let apps_thread = thread::spawn(|| {
-        let mut applications = std::fs::read_dir("/usr/share/applications")?
-            .into_iter()
-            .map(|file| file.unwrap().path())
+        let user_home_app_dir = format!("{}/.local/share/applications", env!("HOME"));
```

Comment:
`env!("HOME")` is treated as a compile time constant thus can't be used in this context.

 I think the cleanest solution is to combine base dirs from:

- `XDG_CONFIG_HOME` (fallback to `$HOME/.local/share`),
- `XDG_DATA_DIRS` (fallback to `[/usr/share, /usr/local/share]`).

This way we can achieve compliance with freedesktop spec:

https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html